### DISCUSSION TITLE
n_ctx inconsistencies (same model, different platforms)

### DISCUSSION THREAD

**Discussion by TheDarkTrumpet** [oobabooga/text-generation-webui#4859] [Q&A] [answered]
Hi All,

I'm struggling with a challenge in text-generation-webui over two platforms.  I have the project, both updated, installed on OSX and in Linux (Debian, to be specific).

Both systems work fine, so I don't have problems executing models, but I've noticed that the n_ctx max is different depending on the host system.

For example, on my mac, I can select a n_ctx to 32k, and on my Linux box, I can only select up to 16k.  I have a feeling I'm missing something simple here, or something I did on the mac, to allow a larger context size.  But I'm unsure what I'm doing incorrect.

The model I'm using for both is: phind-codellama-34b-v2.Q4_K_M.gguf, from TheBloke.

Mac View:

![image](https://github.com/oobabooga/text-generation-webui/assets/92734/5a344b92-0686-422d-a3fc-5b39fc6156aa)

Linux View:

![image](https://github.com/oobabooga/text-generation-webui/assets/92734/7ab3c07d-529c-481d-ad48-bfd6367abe70)

Thanks for any help anyone can provide on what's going on.

**Comment by TheLounger** [comment, marked as answer]
Not sure why you have this inconsistency, but you can just set those values in `models/config-user.yaml` and don't touch the UI after reloading. Example:
```yaml
34B-GGUF__Nous-Capybara-200K$:
  n_gpu_layers: 128
  threads: 32
  n_ctx: 32_768
  max_seq_len: 32_768

**Reply by mongolu** [reply]
Why the underscores?

**Reply by TheDarkTrumpet** [reply]
Sorry for the late reply.  That was my original idea was to update the config to do it and just avoid touching it from model screen (or better yet loading it automatically, or through a script).  I haven't tested how that operates yet.  But I am quite confident that's an option.

@mongolu - I don't think the underscores are required.  When I did this (originally before posting here), I didn't use underscores and it read just fine.  But appears it's optional to denote a "thousands" mark.

I'll see if i can spend time over the holidays digging into why the UI won't allow it higher.  While the workaround does work, the inconsistency is just odd.  To be fair, it's rare that I need 32k context anyways, but since I run this on an RTX6000 ADA, it can be helpful to have the extra context in certain situations.

Thanks again for replying, sorry for the late reply.
